src/test/java/core/basesyntax/service/impl/ReportServiceImplTest.java

```diff
@@ -0,0 +1,77 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.ReportService;
+import core.basesyntax.service.operation.BalanceOperationHandler;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.service.operation.PurchaseOperationHandler;
+import core.basesyntax.service.operation.ReturnOperationHandler;
+import core.basesyntax.service.operation.SupplyOperationHandler;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ReportServiceImplTest {
+    private static final List<String> ONLY_HEADER_REPORT = new ArrayList<>(
+            List.of("fruit,quantity"));
+    private static final List<String> FULL_REPORT = new ArrayList<>(
+            List.of("fruit,quantity",
+                    "apple,90"));
+    private static List<FruitTransaction> fruitTransactions;
+    private static ReportService reportService;
+    private static OperationStrategy operationStrategy;
+
+    @BeforeClass
+    public static void setup() {
```

Comment:
Better create reports in different files and get them for the test, instead of making so big setup for tests.